_meta/ecs-migration.yml

```diff
@@ -0,0 +1,148 @@
+# based on this information in an automated way and to keep track of all changes which were applied.
+#
+# The format of the file is as following:
+#
+# - from: source-field-in-6.x
+#   to: target-filed-in-ECS
+#   # Alias field is useful for fields where there is a 1-1 mapping from old to new
+#   alias: true-if-alias-is-required-in-6x (default is true)
+#   # Copy to is useful for fields where multiple fields map to the same ECS field
+#   copy_to: true-if-field-should-be-copied-to-target-in-6x (default is false)
+
+- from: beat.hostname
+  to: observer.hostname
+
+- from: beat.name
+  to: observer.type
+
+- from: beat.version
+  to: observer.version
+
+- from: context.service.agent.name
+  to: agent.name
+
+- from: context.service.agent.version
+  to: agent.version
+
+- from: context.system.architecture
+  to: host.architecture
+
+- from: context.system.hostname
+  to: host.hostname
+
+- from: context.system.ip
+  to: host.ip
+
+- from: context.system.platform
+  to: host.os.platform
+
+- from: context.request.method
+  to: http.request.method
+
+- from: context.request.http_version
+  to: http.version
+
+- from: context.tags
+  to: labels
+  alias: false
+  copy_to: true
+
+- from: context.process.argv
+  to: process.args
+
+- from: context.process.pid
+  to: process.pid
+
+- from: context.process.ppid
+  to: process.ppid
+
+- from: context.process.title
+  to: process.title
+
+  # not in ECS
+- from: context.service.environment
+  to: service.environment
+
+  # not in ECS
+- from: context.service.framework.name
+  to: service.framework.name
+
+  # not in ECS
+- from: context.service.framework.version
+  to: service.framework.version
+
+  # not in ECS
+- from: context.service.language.name
+  to: service.language.name
+
+  # not in ECS
+- from: context.service.language.version
+  to: service.language.version
+
+- from: context.service.name
+  to: service.name
+
+  # not in ECS
+- from: context.service.runtime.name
+  to: service.runtime.name
+
+  # not in ECS
+- from: context.service.runtime.version
+  to: service.runtime.version
+
+- from: context.service.version
+  to: service.version
+
+- from: context.request.url.full
+  to: url.full
+
+- from: context.request.url.hash
+  to: url.fragment
+
+- from: context.request.url.hostname
+  to: url.domain
+
+- from: context.request.url.pathname
+  to: url.path
+
+- from: context.request.url.port
+  to: url.port
+  alias: false
+  copy_to: true
+
+- from: context.request.url.raw
+  to: url.original
+
+- from: context.request.url.search
+  to: url.query
+
+- from: context.request.url.protocol
+  to: url.scheme
+  alias: false
+  copy_to: true
+
+- from: context.response.finished
+  to: http.response.finished
+
+- from: context.response.status_code
+  to: http.response.status_code
+
+- from: context.user.email
+  to: user.email
+
+- from: context.user.id
+  to: user.id
+
+- from: context.user.username
+  to: user.name
+
+- from: context.user.ip
+  to: client.ip
```

Comment:
Defining the `user.ip` as `client.ip` - would that not also imply that `user` should be nested under `client` instead of being on top level?

Reply:
the other user fields, `id, email, name` don't fit under client.  ECS team has provided guidance to proceed as is while they discuss whether to change that recommendation.

Reply:
The IP pairs in ECS have really been defined in terms of network monitoring and access log analysis. The APM/RUM usage hasn't been explored as much.

I do think `client.ip` makes cognitive sense here, when you see the field name it's pretty clear what it is. I don't think there's a need to necessarily fill out the other side of the pair (`server.ip`), despite what the spec says.

So that wasn't your actual question, but I wanted to make sure I mentioned this here :-)

Yeah nesting user at `client.user` would make sense, I think. It's one of the expected places for `user`.

Reply:
I am good either way, just wanted to make sure it was not overlooked.